src/lib/PreviewError.js

```diff
@@ -0,0 +1,29 @@
+class PreviewError extends Error {
+    /** @property {string} - English error code */
+    code;
+
+    /** @property {Object} - Optional error details, will contain the native error under `originalError` if available */
+    details;
+
+    /** @property {string} - Translated error message if error is shown to user */
+    message;
+
+    /**
+     * [constructor]
+     *
+     * @param {string} code - English error code
+     * @param {string} [message] - Error message, should be translated if displayed to the user
+     * @param {Object} [details] - Optional error details
+     * @param {Object} [details.originalError] - If error originates from a native error, this will refer to that error
+     * @return {PreviewError} PreviewError instance
+     */
+    constructor(code, message, details) {
+        super();
+
+        this.code = code;
+        this.message = message || code;
```

Comment:
Should we use the generic message as a fallback here? This could cleanup logic in the error viewer.

Reply:
Good idea.